src/main/scala/io/qbeast/spark/table/IndexedTable.scala

```diff
@@ -540,7 +541,10 @@ private[table] class IndexedTableImpl(
           val (dataExtended, tableChanges) =
             DoublePassOTreeDataAnalyzer.analyzeOptimize(data, indexStatus)
 
-          val newFiles = dataWriter.write(tableID, schema, dataExtended, tableChanges)
+          val newFiles = dataWriter
+            .write(tableID, schema, dataExtended, tableChanges)
+            .map(addFile => addFile.asInstanceOf[AddFile].copy(dataChange = false))
```

Comment:
Pattern matching is the more appropriate and idiomatic approach in scala than casting with `asInsatanceOf:`
```
.collect {
  case addFile: AddFile => addFile.copy(dataChange = false)
}
```